### PR TITLE
WIP: factory pad models

### PR DESCRIPTION
**Pull request by Anarchid**
No scripting is coding is in scope of this PR; this is just models and perhaps unit files as necessary for the models to be verifiably working in game.

- [x] Cloak
- [x] Shield
- [x] Rover
- [x] Hover
- [x] Tank
- [x] Gunship
- [x] Plane
- [x] Spider
- [x] JJ
- [x] Amph
- [x] Ship
- [ ] Proper decals with AO's
- [ ] Corpses

### CODE REVIEW

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
This one will need some kind of a teamcolor solution. I guess i'll readd the floor grill. 
image

File: Objects3d/pad_tank.dae.lua [hunk 1]
```diff
@@ -0,0 +1,10 @@
+return {
+	radius = 30.00,
+	height = 45.00,
+	midpos = {0.00, 15.00, 0.00},
+	mins = {-65.00,  0.00, -50},
+    maxs = { 65.00, 50.00,  62},
+	tex1 = "factorytank_1.png",
```
Comment (GoogleFrog) [inline Objects3d/pad_tank.dae.lua:7]: Whitespace.

File: units/pad_rover.lua [hunk 2]
```diff
@@ -0,0 +1,101 @@
+return { padrover = {
```
Comment (GoogleFrog) [inline units/pad_rover.lua:1]: File name is incorrect.

File: units/pad_rover.lua [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  unitname                      = [[padrover]],
+  name                          = [[Rover Pad]],
+  description                   = [[Produces Rovers, Builds at 10 m/s]],
+  acceleration                  = 0,
+  brakeRate                     = 0,
+  buildCostMetal                = Shared.FACTORY_COST,
```
Comment (GoogleFrog) [inline units/pad_rover.lua:7]: Shared.FACTORY_PAD_COST

**Comment by GoogleFrog** [comment]
The pads look fine, except perhaps the Tank Pad could have coloured wings to make it less square. I assume that fixing footprints and collision volumes will be a task for someone else.

All non-air pads should have footprints that are impassible rectangles at one end and a 3x3 or 4x4 passable area as the construction zone. This simplifies pathfinding. See the Spider factory for a larger example.

**Comment by Anarchid** [comment]
>  I assume that fixing footprints and collision volumes will be a task for someone else.

Or at least for another PR; i want to get done with the kitbashing before looking at any implementation details.

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
image

**Comment by sprunk** [comment]
Air pad lacks teamcolor.

**Comment by Anarchid** [comment]
Dang, you're right. Will do something similar to tank i guess.

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
image

**Comment by sprunk** [comment]
Is the teamcolored circle a part of the mesh? It would probably be ideal to make the main airfac's buildpad look the same.

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
> Is the teamcolored circle a part of the mesh? It would probably be ideal to make the main airfac's buildpad look the same.

Yes. I cannot touch the s3o file though.

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
image

**Comment by Anarchid** [comment]
Only shield remains now of actual models. I think i'll do a reverse kitbash here: make the pad turret as i want the actual shieldfac to look, and then eventually use that as a basis for full shieldfac model replacement.

**Comment by Anarchid** [comment]
Ok no need to do shieldfac: https://github.com/ZeroK-RTS/Zero-K/commit/190abe9df94ec758bde2f34bec5ef78507faf7c9

**Comment by Anarchid** [comment]
Will leave corpses, decal tunings, and scripts for a separate issue.

**Pull request merged**